Dockerfile

```diff
@@ -1,49 +1,70 @@
 ARG LLVM_VERSION=800
 ARG ARCH=amd64
+ARG UBUNTU_VERSION=18.04
+ARG DISTRO_BASE=ubuntu${UBUNTU_VERSION}
+ARG BUILD_BASE=ubuntu:${UBUNTU_VERSION}
 ARG LIBRARIES=/opt/trailofbits/libraries
-ARG DISTRO_BASE=ubuntu18.04
+
+# Will copy remill installation from here
+FROM trailofbits/remill:llvm${LLVM_VERSION}-${DISTRO_BASE}-${ARCH} as remill
+
+# Additional runtime dependencies go here
+FROM ${BUILD_BASE} as base
+ARG UBUNTU_VERSION
+RUN apt-get update && \
+    if [ "${UBUNTU_VERSION}" = "20.04" ] ; then \
+        apt-get install -qqy --no-install-recommends libtinfo6 ; \
+    else \
+        apt-get install -qqy --no-install-recommends libtinfo5 ; \
+    fi && \
+    apt-get install -qqy --no-install-recommends libz3-4 && \
```

Comment:
So we should package z3 in trailofbits-libraries. We had some version specific issues before, such as the official version not being new enough or requiting OpenMP. 

If we *already* need trailofbits-libraries to build then lets not install this. If we *do not* need trailofbits-libraries, this may be a way to save some space in the docker image.

Reply:
Thank you for the explanation. I just removed this and it should be a bit better now. I had forgotten to test without this before when I was running into other issues. 👍

Reply:
The only thing in the Docker image should be Anvill and Remill (for the semantics files)